idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1333,6 +1340,12 @@ struct RespondQueryTaskCompletedRequest {
   40: optional string errorMessage
 }
 
+struct WorkflowQueryResult {
+  10: optional QueryResultType resultType
+  20: optional binary queryAnswer
+  30: optional string errorMessage
```

Comment:
Should't it be consistent with activity failure which contains both error type and binary details?

Reply:
I think the reason reason ActivityFailure has both is because ActivityReturns errors that look like this 

```
// NewCustomError create new instance of *CustomError with reason and optional details.
// Use CustomError for any use case specific errors that cross activity and child workflow boundaries.
func NewCustomError(reason string, details ...interface{}) *CustomError {
	return internal.NewCustomError(reason, details...)
}
```

Query does not return such an error. If you look at the RespondQueryTaskCompletedRequest it looks like 

```
struct RespondQueryTaskCompletedRequest {
  10: optional binary taskToken
  20: optional QueryTaskCompletedType completedType
  30: optional binary queryResult
  40: optional string errorMessage
}
```

Reply:
I know. I think currently we have to reuse queryResult to pass error details which is not very clean design.